src/Giraffe/Routing.fs

```diff
@@ -10,37 +10,97 @@ open FSharp.Control.Tasks.V2.ContextInsensitive
 open Giraffe.FormatExpressions
 
 // ---------------------------
-// Private sub route helper functions
+// Sub Routing Feature
 // ---------------------------
 
-[<RequireQualifiedAccess>]
-module SubRouting =
+[<Literal>]
+let private RouteKey = "giraffe_route"
+
+type ISubRoutingFeature =
```

Comment:
Is this interface going to be enough? Afaik F# isn't going to support DIM implementations for a while, so versioning it could become nasty.

Reply:
Continuing, isn't this interface so specific that there's not a lot of value in it? If so you could argue that you could better represent this as a single property on a concrete class and be done with it.